QEfficient/compile/qnn_compiler.py

```diff
@@ -0,0 +1,386 @@
+            :output_dir (str): Path to store generated Binaries (qpc_path/qpcs/).
+            :model (str): Path to the <qnn_model_name.so> file containing a QNN network.
+            :dlc_path (str): Path to DLC file generated by QNN-Convertor.
+            :config_file(str): Path to created qnn_compiler_config.json containing qnn_compile_backend.json & shared_library_path.
+
+        Configurable parameters:
+            :log_level(str): ``Configurable`` Default(error).
+
+        CONTEXT_BIN_ARGS_EXTENSION passed in qnn_config.json is appended to the command created.
+
+        """
+        binary_gen_tool = QnnConstants.QNN_CONTEXT_BIN.format(self.qnn_sdk_path, self.qnn_target)
+        backend_lib = QnnConstants.QNN_CONTEXT_LIB_BACKEND.format(self.qnn_sdk_path, self.qnn_target)
+        model_lib = QnnConstants.QNN_CONTEXT_LIB_MODEL.format(self.qnn_sdk_path, self.qnn_target)
+        config_file_path = self.create_qnn_compiler_config_json()
+
+        cmd = (
+            f"{binary_gen_tool} --binary_file {QnnConstants.CONTEXT_BIN_NAME} "
+            f"--backend_binary {QnnConstants.CONTEXT_BIN_QPC_NAME} "
+            f"--output_dir {self.qnn_binary_dir} "
+            f"--backend {backend_lib} "
+            f"--model {model_lib} "
+            f"--dlc_path {self.dlc_model_path} "
+            f"--config_file {config_file_path} "
+        )
+
+        if self.qnn_config and QnnConstants.CONTEXT_BIN_ARGS_EXTENSION_STR in self.qnn_config:
+            if "--log_level " not in self.qnn_config[QnnConstants.CONTEXT_BIN_ARGS_EXTENSION_STR]:
+                cmd += f"--log_level {QnnConstants.LOG_LEVEL} "
+            cmd += self.qnn_config[QnnConstants.CONTEXT_BIN_ARGS_EXTENSION_STR]
+        else:
+            cmd += f"--log_level {QnnConstants.LOG_LEVEL} "
+
+        return cmd
+
+    def quantize(self):
+        raise NotImplementedError("QNN Quantization is not supported")
+
+    def execute(self):
+        raise NotImplementedError("QNN Execution is not supported")
+
+    def generate_profiling(self):
+        raise NotImplementedError("QNN profiling is not supported")
+
+
+def create_json(filePath: str, jsonData: object):
+    """
+    Creates a JSON file with provided JSON data.
+
+    ``Mandatory`` Args:
+        :filePath (str): JSON File to be created.
+        :jsonData (object): JSON Data Object to be populated inside the created file.
+
+    """
+    try:
+        with open(filePath, "w") as file:
+            json.dump(jsonData, file, indent=4)
+    except Exception as e:
+        print(f"Failed to create JSON File {filePath}: {e}")
+
+
+def compile(
+    onnx_path: str,
+    qpc_path: str,
+    num_cores: int,
+    device_group: Optional[List[int]] = None,
+    aic_enable_depth_first: bool = False,
+    mos: int = -1,
+    batch_size: int = 1,
+    prompt_len: int = 32,
+    ctx_len: int = 128,
+    mxfp6: bool = True,
+    mxint8: bool = False,
+    full_batch_size=None,
+    qnn_config: Optional[str] = None,
+    **kwargs,
```

Comment:
kwargs is not utilized anywhere

Reply:
Added check similar to qaic compiler function:
    if kwargs:
        raise NotImplementedError("Can't handle extra compilation args now!")
Let me know if I should I remove kwargs completely

Reply:
Added a warning as per Rishin's suggestion.